src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -300,8 +331,8 @@ internal static SafeSslHandle AllocateSslHandle(SslAuthenticationOptions sslAuth
                 }
                 else
                 {
-
-                    s_clientSslContexts.TryGetValue(protocols, out sslCtxHandle);
+                    var key = new SslContextCacheKey(protocols, sslAuthenticationOptions.CertificateContext?.TargetCertificate.GetCertHash());
```

Comment:
This is a new use of SHA-1, which isn't permissible.

My normal recommendation is "can you just use TargetCertificate.RawDataMemory and ignore the hashing/collision?"

If that's not tenable

```suggestion
                    var key = new SslContextCacheKey(protocols, sslAuthenticationOptions.CertificateContext?.TargetCertificate.GetCertHash(HashAlgorithmName.SHA256));
```